lnd_test.go

```diff
@@ -488,6 +490,128 @@ func testSingleHopInvoice(net *networkHarness, t *harnessTest) {
 	closeChannelAndAssert(t, net, ctxt, net.Alice, chanPoint, false)
 }
 
+func testListPayments(net *networkHarness, t *harnessTest) {
+	ctxb := context.Background()
+	timeout := time.Duration(time.Second * 5)
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
```

Comment:
The timeout contexts are _absolute_, so you'll need to create a new timeout context for each RPC call in order to ensure we don't introduce some flakiness due to the same timeout context being used throughout the test. 

As it stands now, if the entire test doesn't complete in 5 seconds (although it likely should), then the test will fail due to expiration of a context.